documentation/docs/libraries/nodejs/exchange_guide.md

```diff
@@ -0,0 +1,238 @@
+
+<CodeBlock className="language-javascript">
+  {generate_address}
+</CodeBlock>
+
+Every account can have multiple addresses. Addresses are represented by an `index` which is incremented (by 1) every time a new address is created. The addresses are accessible via `account.listAddress()`: 
+
+```javascript
+    const addresses = account.listAddresses()
+
+    console.log('Need a refill? Send it to this address:', addresses[0])
+```
+You can fill the address with Tokens with the [Faucet](https://faucet.testnet.shimmer.network/) to test it.
+
+Addresses are of two types, `internal` and `public` (external):
+
+* Each set of addresses are independent from each other and has an independent `index` id.
+* Addresses that are created by `account.generateAddress()` are indicated as `internal=false` (public).
+* Internal addresses (`internal=true`) are called `change` addresses and are used to send the excess funds to them.
+* The approach is also known as a *BIP32 Hierarchical Deterministic wallet (HD Wallet)*.
+
+### 4. Check the Account Balance
+
+Get the available account balance across all addresses of the given account:
+
+<CodeBlock className="language-javascript">
+  {check_balance}
+</CodeBlock>
+
+### 5. Listen to Events
+
+The `Wallet.rs` library supports several events for listening. As soon as the given event occurs (which usually happens during syncing), a provided callback is triggered.
+
+Below is an example for listening to new output events:
+
+<CodeBlock className="language-javascript">
+  {listen_events}
+</CodeBlock>
+
+Example output:
+
+```bash
+TODO: update
+data: {
+  accountId: '0',
+  address: {
+    address: 'atoi1q9c6r2ek5w2yz54en78m8dxwl4qmwd7gmh9u0krm45p8txxyhtfry6apvwj',
+    balance: 20000000,
+    keyIndex: 0,
+    internal: false,
+    outputs: [ [Object], [Object] ]
+  },
+  balance: 20000000
+}
+```
+
+`accountId` can then be used to identify the given account via `AccountManager.getAccount(accountId)`.
+
+### 6. Enable Withdrawals
+
+Sending coins:
+
+<CodeBlock className="language-javascript">
+  {send_amount}
+</CodeBlock>
+
+The full function signature is `Account.sendAmount(outputs[, options])`.
+
+Default options are fine and successful; however, additional options can be provided, such as `remainderValueStrategy`:
+
+* `changeAddress`: Send the remainder value to an internal address.
+* `reuseAddress`: Send the remainder value back to its original address.
+
+```json
+TransactionOptions {
+    remainderValueStrategy?: RemainderValueStrategy;
+    taggedDataPayload?: ITaggedDataPayload;
+    customInputs?: string[];
+}
+```
+
+The `Account.send()` function returns a `transaction` with it's id. The `blockId` can be used later for checking a confirmation status. Individual transactions related to the given account can be obtained via the `account.listTransactions()` function.
```

Comment:
```suggestion
The [`Account.send()`](../../reference/nodejs#sendaddress-amount-options) function returns a `transaction` with it's id. The `blockId` can be used later for checking a confirmation status. You can obtain individual transactions related to the given account using the `account.listTransactions()` function.
```

Reply:
Updated the text part, but reference is not updated yet, so will not link to it